### PR TITLE
docs(framework) Add Flower architecture explanation

### PR DESCRIPTION
**Pull request by chongshenng**
Add the Flower explainer page and step-by-step guide.

### CODE REVIEW

File: doc/source/explanation-flower-next.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+Messages
+~~~~~~~~
+
+Message (|message_link|_) is an object that Flower uses to carry information between ServerApp and ClientApp. This information can be a model the ServerApp wants to federate, metrics the ClientApp is pushing back to the ServerApp via the SuperLink, and anything in between. The design of Messages and how they are handled by Flower ensures that a Message sent by the ServerApp looks exactly the same when received by the ClientApp (and vice versa). This ensures a more unified and smoother developer experience.
+
+Context
+~~~~~~~
+
+Context (|context_link|_) is another useful object that we introduced in Flower Next. For every execution of `ServerApp`, i.e. a run, it carries the record and messages that is unique to the run. Each time a SuperNode runs a ClientApp within the run, the same Context object is exposed to the ClientApp, allowing the ClientApp to persist throughout the duration of the run. Context can be used as a temporary buffer for executing additional tasks, like `mods`.
```
Comment (mohammadnaseri) [inline doc/source/explanation-flower-next.rst:93]: I would reference (https://flower.ai/docs/framework/how-to-use-built-in-mods.html) after mentioning mods.

File: doc/source/_static/flower-infrastructure-applications.png
[the remark this answers is not in the document]
Comment (chongshenng) [reply doc/source/_static/flower-infrastructure-applications.png:1]: 1. You've a point about the flag. Let's remove that.
2. That's true. Let's move that. 
3. Indeed, let's see how we can add that, maybe to the legend.

File: doc/source/conf.py [hunk 2]
```diff
@@ -265,6 +263,8 @@ def find_test_modules(package_path):
     "ref-api/flwr.simulation.run_simulation_from_cli": "index.html",
     "contributor-how-to-create-new-messages": "index.html",
     "example-jax-from-centralized-to-federated": "tutorial-quickstart-jax.html",
+    "architecture": "index.html",
+    "contributor-explanation-architecture.html": "index.html",
```
Comment (danieljanes) [inline doc/source/conf.py:267]:
```suggestion
    "architecture": "explanation-flower-architecture.html",
    "contributor-explanation-architecture.html": "explanation-flower-architecture.html",
```

**Review by danieljanes — COMMENTED** [review]
The removed contributor doc also needs to be removed from `index.rst`

**Review by danieljanes — APPROVED** [review]

**Pull request merged**